### DISCUSSION TITLE
Can the HexCtrl methods be wrapped to C style exported functions?

### DISCUSSION THREAD

**Discussion by mrfearless** [jovibor/HexCtrl#86] [Q&A] [answered]
I think your HexCtrl is excellent. I mainly use assembler x86/x64 masm/uasm with win32 api calls, and was thinking of using your control, but it might be easier for myself to do so if the hex control functions and methods are accessible via c exported function calls,  essentially wrapper functions to call the methods etc, that way the use of the HexCtrl can be accessible to other program languages. Also is it possible to compile the HexCtrl as static library instead of using as a dll? Anyways love your work. Thanks.

**Comment by jovibor** [comment, marked as answer]
Thank you for kind words.

Since **HexCtrl** is developed in *C++20*, rewriting it to be available as a pure old *C* interface would be a big step back imo.
But I don't see any limitations here, if you for some reason need such a *C* style access you can always make a wrapper yourself since all the sources are available.

> Also is it possible to compile the HexCtrl as static library instead of using as a dll?

Static library is basically a *.dll* baked into an executable. I didn't have a necessity to create one, but since as a *.dll* **HexCtrl** works fine for a long time I don't see any reason why it wouldn't as a static library. The difference is basically in one project setting.

If you have any other questions feel free to ask.
